### PR TITLE
cudaPackages.nccl-tests: 2.13.6 -> 2.13.8

### PR DESCRIPTION
**Pull request by ConnorBaker**
## Description of changes

- Version bump: 2.13.6 -> 2.13.8
- Add `updateScript`
- Add CUDA team as maintainers
- Enable building on all linux platforms
- Fixes a possible error by which `nccl-tests` is provided `cudaPackages` from the top-level `pkgs` definition, and not from the `cudaPackages` of the enclosing scope.
  - As an example, if `cudaPackages` is the default version of CUDA packages in Nixpkgs and you tried to build `cudaPackages_12_2.nccl-tests`, `nccl-tests` wouldn't receive `cudaPackages_12_2` -- no, instead it would get `cudaPackages` from the top level scope!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ConnorBaker** [comment]
Result of `nixpkgs-review pr 266078 --extra-nixpkgs-config '{
      allowUnfree = true;
      cudaSupport = true;
      cudaCapabilities = [ "8.9" ];
      packageOverrides = pkgs: { cudaPackages = pkgs.lib.recurseIntoAttrs pkgs.cudaPackages_11_8; };
    }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cudaPackages.nccl-tests</li>
  </ul>
</details>

**Review by SomeoneSerge — APPROVED** [review]
(the update script works too)

File: pkgs/development/libraries/science/math/nccl/tests.nix [hunk 1]
```diff
@@ -46,12 +54,18 @@ cudaPackages.backendStdenv.mkDerivation (finalAttrs: {
     cp -r build/* $out/bin/
   '';
 
+  # Run the update script with: `nix-shell maintainers/scripts/update.nix --argstr package cudaPackages.nccl-tests`
```
Comment (jmillerpdt) [inline pkgs/development/libraries/science/math/nccl/tests.nix:57]: Is this documentation for yourself or intentional in the PR?  A quick search for it shows documentation in `doc/stdenv`.

**Review by jmillerpdt — APPROVED** [review]

File: pkgs/development/libraries/science/math/nccl/tests.nix (hunk 1 above)
Comment (ConnorBaker) [reply pkgs/development/libraries/science/math/nccl/tests.nix]: My bad, that was just for myself! Removed in latest.

**Pull request merged**